src/_pytest/terminal.py

```diff
@@ -878,8 +878,10 @@ def pytest_sessionfinish(
     def pytest_terminal_summary(self) -> Generator[None, None, None]:
         self.summary_errors()
         self.summary_failures()
+        self.summary_xfailures()
         self.summary_warnings()
         self.summary_passes()
+        self.summary_xpasses()
```

Comment:
`summary_xfailures` and `summary_failures` are nearly identical, same for `summary_xpasses` and `summary_passes`. Can we refactor this so we can have just one function for each, `summary_failures` and `summary_passes`, which receives the parameters needed to handle the normal vs "x" variants?

Reply:
good idea, heading to a commit soon